orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTaskSpec.groovy

```diff
@@ -146,15 +146,58 @@ class MonitorKatoTaskSpec extends Specification {
     "about to fail"      | MonitorKatoTask.TASK_NOT_FOUND_TIMEOUT     |  true         | 'N/A'
   }
 
+  @Unroll
+  def "should skip replicas when initial timeout is exceeded #desc"() {
+    given:
+    def ctx = [
+      "kato.last.task.id": new TaskId(taskId)
+    ]
+    if (previousRetry) {
+      ctx.put('kato.task.firstNotFoundRetry', now.minusMillis(elapsed).toEpochMilli())
+    } else {
+      if (previousRetryFlag != null) {
+        ctx.put('kato.task.firstNotFoundRetry', previousRetryFlag)
+      }
+    }
+
+    def stage = new Stage(Execution.newPipeline("orca"), "whatever", ctx)
+    task.kato = Stub(KatoService) {
```

Comment:
This sort of thing might make more sense as a `Mock` that you then configure and assert calls against in your `then` block